File: CS/WindowsApplication1/GridExportHelper.cs
using System;
using System.Drawing;
using DevExpress.XtraGrid;
using System.IO;
using DevExpress.Utils;
using DevExpress.XtraGrid.Views.Grid;

namespace WindowsApplication1
{
    public class GridExportHelper
    {
        private  GridControl _GridControl;
        MemoryStream layoutStream = new MemoryStream();


        public static void ExportGridControl(GridControl gc, string fileName)
        {

            new GridExportHelper().InternalExportGridControl(gc, fileName);
        }

        private GridView ExportedView
        {
            get
            {
                return _GridControl.MainView as GridView;
            }
        }
        private void InternalExportGridControl(GridControl gc, string fileName)
        {
            _GridControl = gc;
            try
            {
                LockUpdate();
                SaveLayout();
                SubscribeEvents(ExportedView);
                CustomizeGridViewBeforeExport(ExportedView);
                DoExport(fileName);
            }
            finally
            {
                UnsubscribeEvents(ExportedView);
                RestoreLayout();
                UnlockUpdate();
            }
        }

        private void LockUpdate()
        {
            _GridControl.BeginUpdate();

        }
        private void UnlockUpdate()
        {
            _GridControl.EndUpdate();
        }
        private void RestoreLayout()
        {
            layoutStream.Position = 0;
            ExportedView.RestoreLayoutFromStream(layoutStream, OptionsLayoutBase.FullLayout);
        }
        private void SaveLayout()
        {
            ExportedView.SaveLayoutToStream(layoutStream, OptionsLayoutBase.FullLayout);
        }

        private void CustomizeGridViewBeforeExport(GridView view)
        {
            view.OptionsView.ColumnAutoWidth = false;
            view.BestFitColumns();
            view.OptionsPrint.AutoWidth = false;
        }


        void view_CustomColumnDisplayText(object sender, DevExpress.XtraGrid.Views.Base.CustomColumnDisplayTextEventArgs e)
        {
            e.DisplayText = String.Format("[{0}]", e.DisplayText);
        }

        private void SubscribeEvents(GridView view)
        {
            view.CustomColumnDisplayText += view_CustomColumnDisplayText;
        }
        private void UnsubscribeEvents(GridView view)
        {
            view.CustomColumnDisplayText -= view_CustomColumnDisplayText;
        }

        private void DoExport(string fileName)
        {
            _GridControl.ExportToXls(fileName);
        }
    }
}

File: VB/GridExportHelper.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports DevExpress.XtraGrid
Imports System.IO
Imports DevExpress.Utils
Imports DevExpress.XtraGrid.Views.Grid

Namespace WindowsApplication1
	Public Class GridExportHelper
		Private _GridControl As GridControl
		Private layoutStream As New MemoryStream()


		Public Shared Sub ExportGridControl(ByVal gc As GridControl, ByVal fileName As String)

			CType(New GridExportHelper(), GridExportHelper).InternalExportGridControl(gc, fileName)
		End Sub

		Private ReadOnly Property ExportedView() As GridView
			Get
				Return TryCast(_GridControl.MainView, GridView)
			End Get
		End Property
		Private Sub InternalExportGridControl(ByVal gc As GridControl, ByVal fileName As String)
			_GridControl = gc
			Try
				LockUpdate()
				SaveLayout()
				SubscribeEvents(ExportedView)
				CustomizeGridViewBeforeExport(ExportedView)
				DoExport(fileName)
			Finally
				UnsubscribeEvents(ExportedView)
				RestoreLayout()
				UnlockUpdate()
			End Try
		End Sub

		Private Sub LockUpdate()
			_GridControl.BeginUpdate()

		End Sub
		Private Sub UnlockUpdate()
			_GridControl.EndUpdate()
		End Sub
		Private Sub RestoreLayout()
			layoutStream.Position = 0
			ExportedView.RestoreLayoutFromStream(layoutStream, OptionsLayoutBase.FullLayout)
		End Sub
		Private Sub SaveLayout()
			ExportedView.SaveLayoutToStream(layoutStream, OptionsLayoutBase.FullLayout)
		End Sub

		Private Sub CustomizeGridViewBeforeExport(ByVal view As GridView)
			view.OptionsView.ColumnAutoWidth = False
			view.BestFitColumns()
			view.OptionsPrint.AutoWidth = False
		End Sub


		Private Sub view_CustomColumnDisplayText(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Base.CustomColumnDisplayTextEventArgs)
			e.DisplayText = String.Format("[{0}]", e.DisplayText)
		End Sub

		Private Sub SubscribeEvents(ByVal view As GridView)
			AddHandler view.CustomColumnDisplayText, AddressOf view_CustomColumnDisplayText
		End Sub
		Private Sub UnsubscribeEvents(ByVal view As GridView)
			RemoveHandler view.CustomColumnDisplayText, AddressOf view_CustomColumnDisplayText
		End Sub

		Private Sub DoExport(ByVal fileName As String)
			_GridControl.ExportToXls(fileName)
		End Sub
	End Class
End Namespace
